semaphore_admission.go

```diff
@@ -0,0 +1,65 @@
+package frankenphp
+
+import (
+	"context"
+
+	"golang.org/x/sync/semaphore"
+)
+
+func acquireSemaphoreWithAdmissionControl(
```

Comment:
nit: could maybe fit on one line?

Reply:
My eyesight has been going the last few months, so I need a bigger font. 

<img width="1896" height="918" alt="Image" src="https://github.com/user-attachments/assets/c81cfb28-aa54-454e-a46b-3d940e3cb2b3" />

vs.

<img width="1967" height="862" alt="Image" src="https://github.com/user-attachments/assets/6d60f367-a610-4f7d-bb26-a6b9b41ce62a" />

The return type is off the edge of the screen.

Reply:
+1 for multiple lines